axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveLineServiceImpl.java

```diff
@@ -0,0 +1,1309 @@
+  public MoveLine generateTaxPaymentMoveLineList(
+      MoveLine customerMoveLine, Invoice invoice, Reconcile reconcile) throws AxelorException {
+    BigDecimal paymentAmount = reconcile.getAmount();
+    BigDecimal invoiceTotalAmount = invoice.getCompanyInTaxTotal();
+    for (InvoiceLineTax invoiceLineTax : invoice.getInvoiceLineTaxList()) {
+
+      TaxLine taxLine = invoiceLineTax.getTaxLine();
+      BigDecimal vatRate = taxLine.getValue();
+      BigDecimal baseAmount = invoiceLineTax.getCompanyExTaxBase();
+      BigDecimal detailPaymentAmount =
+          baseAmount
+              .multiply(paymentAmount)
+              .divide(invoiceTotalAmount, 6, RoundingMode.HALF_EVEN)
+              .setScale(2, RoundingMode.HALF_UP);
+
+      TaxPaymentMoveLine taxPaymentMoveLine =
+          new TaxPaymentMoveLine(
+              customerMoveLine,
+              taxLine,
+              reconcile,
+              vatRate,
+              detailPaymentAmount,
+              Beans.get(AppBaseService.class).getTodayDate());
+
+      taxPaymentMoveLine = taxPaymentMoveLineService.computeTaxAmount(taxPaymentMoveLine);
+
+      customerMoveLine.addTaxPaymentMoveLineListItem(taxPaymentMoveLine);
+    }
+    this.computeTaxAmount(customerMoveLine);
+    return Beans.get(MoveLineRepository.class).save(customerMoveLine);
+  }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public MoveLine reverseTaxPaymentMoveLines(MoveLine customerMoveLine, Reconcile reconcile)
+      throws AxelorException {
+    List<TaxPaymentMoveLine> reverseTaxPaymentMoveLines = new ArrayList<TaxPaymentMoveLine>();
+    for (TaxPaymentMoveLine taxPaymentMoveLine : customerMoveLine.getTaxPaymentMoveLineList()) {
+      if (!taxPaymentMoveLine.getIsAlreadyReverse()
+          && taxPaymentMoveLine.getReconcile().equals(reconcile)) {
+        TaxPaymentMoveLine reverseTaxPaymentMoveLine =
+            taxPaymentMoveLineService.getReverseTaxPaymentMoveLine(taxPaymentMoveLine);
+
+        reverseTaxPaymentMoveLines.add(reverseTaxPaymentMoveLine);
+      }
+    }
+    for (TaxPaymentMoveLine reverseTaxPaymentMoveLine : reverseTaxPaymentMoveLines) {
+      customerMoveLine.addTaxPaymentMoveLineListItem(reverseTaxPaymentMoveLine);
+    }
+    this.computeTaxAmount(customerMoveLine);
+    return Beans.get(MoveLineRepository.class).save(customerMoveLine);
+  }
+
+  @Transactional(rollbackOn = {Exception.class})
+  public MoveLine computeTaxAmount(MoveLine moveLine) throws AxelorException {
+    moveLine.setTaxAmount(BigDecimal.ZERO);
+    if (!ObjectUtils.isEmpty(moveLine.getTaxPaymentMoveLineList())) {
+      for (TaxPaymentMoveLine taxPaymentMoveLine : moveLine.getTaxPaymentMoveLineList()) {
+        moveLine.setTaxAmount(moveLine.getTaxAmount().add(taxPaymentMoveLine.getTaxAmount()));
+      }
+    }
+    return moveLine;
+  }
+
+  @Override
+  public Account getEquivalentAccount(MoveLine moveLine) {
+    if (moveLine.getAccount() == null) return null;
```

Comment:
always use brackets `{}` when using if or for, even if there is only one line.
Use && instead of writing two differents if